packages/wallet/src/PersonalWallet/PersonalWallet.ts

```diff
@@ -480,18 +494,20 @@ export class PersonalWallet implements ObservableWallet {
   }
 
   async submitTx(
-    input: Cardano.Tx | TxCBOR | OutgoingTx,
+    input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs,
     { mightBeAlreadySubmitted }: SubmitTxOptions = {}
   ): Promise<Cardano.TransactionId> {
     const outgoingTx = processOutgoingTx(input);
     this.#logger.debug(`Submitting transaction ${outgoingTx.id}`);
     this.#newTransactions.submitting$.next(outgoingTx);
     try {
       await this.txSubmitProvider.submitTx({
+        ...(outgoingTx.context && { context: { handles: outgoingTx.context } }),
```

Comment:
This reads a little weird, how about
```suggestion
        context: outgoingTx.context,
```

Reply:
I thought we wanted to make it optional... so if it didn't exist, we just wouldn't pass it, so we wouldn't have to modify a lot of the other transactions that consume the outgoingTx. Or that's what I understood you and Daniele agreed in the call?

Reply:
Optional means `T | undefined`, setting it to `undefined` explicitly is fine. I understand there's a slight difference between `{}` and `{context: undefined}`, but both matches the type which is the important thing.

Reply:
2d88f9ab41c0bf941b8126409e8d597fa2713303  should address this